prow/config/config.go

```diff
@@ -362,6 +374,36 @@ func (c *Config) getProwYAML(gc git.ClientFactory, identifier string, baseSHAGet
 	return prowYAML, nil
 }
 
+// getProwYAMLNoDefault is like getProwYAML, but it does __not__ use the
+// information found inside the retrieved *ProwYAML to modify the Config.
+func (c *Config) getProwYAMLNoDefault(gc git.ClientFactory, identifier string, baseSHAGetter RefGetter, headSHAGetters ...RefGetter) (*ProwYAML, error) {
+	if identifier == "" {
+		return nil, errors.New("no identifier for repo given")
+	}
+	if !c.InRepoConfigEnabled(identifier) {
+		return &ProwYAML{}, nil
+	}
```

Comment:
As currently written, we'll cache these results as well. In other words we'll be caching the empty inrepo configs of repos that don't enable inrepo config which will waste cache space and negatively impact eviction behavior.
We should instead only cache inrepo config for repos that enable it. Checking if a repo enables inrepo config is efficient and does not need to be cached, we can check that every request.